### PR TITLE
Rfink/jaxrs clients proxy

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

**Comment by uschi2000** [comment]
@iamthomaspowell

File: service-config/src/main/java/com/palantir/config/service/BasicCredentials.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @Value.Parameter
+    public abstract String username();
+
+    @Value.Parameter
+    public abstract String password();
+
+    @Check
+    protected final void check() {
+        Preconditions.checkArgument(!username().isEmpty(), "Username must not be empty");
+        Preconditions.checkArgument(!password().isEmpty(), "Password must not be empty");
```
Comment (markelliot) [inline service-config/src/main/java/com/palantir/config/service/BasicCredentials.java]: I'm not sure we should enforce non-empty constraints

**Comment by markelliot** [comment]
lgtm besides nit on username/password: my guess is that at least one of these we run into has just a username or just a password, some day.

**Comment by uschi2000** [comment]
I guess that ":", "foo:", and ":bar" are all valid basic auth creds?

**Comment by uschi2000** [comment]
rebasing onto develop.

**Pull request merged**